src/main/java/com/github/cameltooling/lsp/internal/parser/CamelEIPParser.java

```diff
@@ -0,0 +1,51 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+
+public class CamelEIPParser {
+
+	private final TextDocumentItem document;
+
+	public CamelEIPParser(TextDocumentItem textDocumentItem) {
+		this.document = textDocumentItem;
+	}
+
+	public boolean canPutEIP(Position position) {
+		Pattern CHOICE_EIP_PATTERN = Pattern.compile("\\)\\s*[.[c[h[o[i[c[e]?]?]?]?]?]?]?\\(?$", Pattern.MULTILINE);
+		ParserFileHelperUtil util = new ParserFileHelperUtil();
+		String textUntilPosition = util.getTextUntilPosition(document, position);
+
+		return CHOICE_EIP_PATTERN.matcher(textUntilPosition).find();
+	}
+
+	public CompletableFuture<List<CompletionItem>> getCompletions() {
+		return CompletableFuture.completedFuture(
+				List.of(choiceEIPcompletion())
+		);
+	}
+
+	private CompletionItem choiceEIPcompletion() {
+		String newLine = System.getProperty("line.separator");
+		CompletionItem completion = new CompletionItem("Content Based Router");
+		completion.setDocumentation(
```

Comment:
i think providing a bit more details on what is the Content based router with a single sentence would help users. This can be done in another iteration.